proposals/dao/tip_121c.ts

```diff
@@ -78,6 +124,75 @@ const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts,
+  // DAI on PSM should cover the user circulating supply of FEI
+  expect(await contracts.dai.balanceOf(addresses.simpleFeiDaiPSM)).to.be.bignumber.greaterThan(
+    userCirculatingFeiSupply
+  );
+  expect(await contracts.fei.balanceOf(addresses.simpleFeiDaiPSM)).to.equal(0);
+
+  // 3. Detailed mint and redeem e2e tests are in 'simpleFeiDaiPSM.ts'
+  // Tests here are sanity checks only that functions work
+  // Swap
+  const daiWhale = '0x5777d92f208679DB4b9778590Fa3CAB3aC9e2168';
+  await forceEth(daiWhale);
+  const daiWhaleSigner = await getImpersonatedSigner(daiWhale);
+  const initialWhaleDaiBalance = await contracts.dai.balanceOf(daiWhale);
+  const initialPSMDaiBalance = await contracts.dai.balanceOf(addresses.simpleFeiDaiPSM);
+  const amountIn = ethers.constants.WeiPerEther.mul(1_000);
+
+  // Mint 1000 FEI, by transferring in 1000 DAI
+  await contracts.dai.connect(daiWhaleSigner).approve(addresses.simpleFeiDaiPSM, amountIn);
+  await contracts.simpleFeiDaiPSM.connect(daiWhaleSigner).mint(daiWhale, amountIn, amountIn);
+
+  // User should have been minted FEI - 1000 DAI sent to contract, 1000 FEI minted to user
+  expect(await contracts.fei.balanceOf(daiWhale)).to.equal(amountIn);
+  const whaleDaiDiff = initialWhaleDaiBalance.sub(await contracts.dai.balanceOf(daiWhale));
+  expect(whaleDaiDiff).to.equal(amountIn);
+  expect((await contracts.dai.balanceOf(addresses.simpleFeiDaiPSM)).sub(initialPSMDaiBalance)).to.equal(amountIn);
+
+  // Redeem - swap 1000 FEI for 1000 DAI. Contract left with 1000 FEI
+  await contracts.fei.connect(daiWhaleSigner).approve(addresses.simpleFeiDaiPSM, amountIn);
+  await contracts.simpleFeiDaiPSM.connect(daiWhaleSigner).redeem(daiWhale, amountIn, amountIn);
+  expect(await contracts.fei.balanceOf(addresses.simpleFeiDaiPSM)).to.equal(amountIn);
+  expect(await contracts.fei.balanceOf(daiWhale)).to.equal(0);
+  expect(await contracts.dai.balanceOf(daiWhale)).to.equal(initialWhaleDaiBalance);
+
+  // Burn the 1000 FEI held on the contract
+  await contracts.simpleFeiDaiPSM.burnFeiHeld();
+  expect(await contracts.fei.balanceOf(addresses.simpleFeiDaiPSM)).to.equal(0);
+
+  //////////////////////    TRIBE REDEEMER    /////////////////////
+  // 4. Verify Tribe Redeemer contract deploy params
+  expect(await contracts.tribeRedeemer.redeemBase()).to.equal(REDEEM_BASE);
+  expect(await contracts.tribeRedeemer.redeemedToken()).to.equal(addresses.tribe);
+  const expectedTokensReceived = [addresses.steth, addresses.lqty, addresses.fox, addresses.dai];
+  const actualTokensReceived = await contracts.tribeRedeemer.tokensReceivedOnRedeem();
+  expect(actualTokensReceived.length).to.equal(expectedTokensReceived.length);
+  expect(actualTokensReceived).to.contain(actualTokensReceived[0]);
+  expect(actualTokensReceived).to.contain(actualTokensReceived[1]);
+  expect(actualTokensReceived).to.contain(actualTokensReceived[2]);
+  expect(actualTokensReceived).to.contain(actualTokensReceived[3]);
+
```

Comment:
These need to be logged and compared against a hardcoded min amount as well imo

Reply:
Have logged. What do you mean compared against a hardcoded min amount? Just addresses

There are minimum balance checks a few lines down